pactman/verifier/verify.py

```diff
@@ -90,14 +107,30 @@ def service_PATCH(self, service_url):
         r = requests.patch(self._get_url(service_url), json=self._request_payload, headers=self._request_headers)
         return self.response.verify(r)
 
-    def setup(self, setup_url):
+    def set_provider_state(self, provider_setup):
+        if self.providerState is not None:
+            log.debug(f'Setting up provider state {self.providerState!r}')
+            return provider_setup(self.providerState)
```

Comment:
Could we move the `return` to the next line:

```python
provider_setup(self.providerState)
return
```

Otherwise it looks like there could be a bug since we return the result of `provider_setup` for `providerState` but not `providerStates`.